### DISCUSSION TITLE
Starting sumantrapdf(portable version) from taskbar pin then opening pdfs in explorer results in creation of new instance, even with reuseinstance set to true

### DISCUSSION THREAD

**Discussion by friday2029** [sumatrapdfreader/sumatrapdf#3980] [Q&A]
As stated in the title if I launch sumatrapdf(portable version) from my taskbar pin, opening pdfs in file explorer will result in a new instance of the application. I do have reuseinstance set to true and it successfully reuse the instance if I launch sumatra initially by opening a pdf from explorer. It's not a major issue to bypass but it is not ideal. I think it might be noteworthy that I am using windows 11, since this was never an issue before I upgraded my pc and switched to 11. I am also using explorer patcher to get a right aligned task bar but I don't think that's effecting this situation.

**Comment by GitHubRulesOK** [comment]
starting position ![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/08814b9e-41ff-4e5f-bebe-fc4953027f30)
select file 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/89cbbbf8-aed7-4fea-85ef-0e228db6b3b6)
one instance 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/a839e1de-1e59-4c44-97cb-a61f02563380)
reselect instance 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/8d9f0127-a44b-471b-a5df-c3bfbf66dcc8)
reuse instance only same instance is available 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/5aa1e8f5-36dc-4ae5-92df-8d362025ab80)

set reuse instance false **then with the tabs active** the setting is normally ignored (still opens existing instance), which is historic behaviour.

however with tabs off it should be different. as historically reuse-instance was developed before tabs 
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/92e9d442-b8f7-4bc5-986a-5d0b2302193b)

**Reply by friday2029** [reply]
Sorry I don't quite understand what you are trying to say. I am using tabs, usually after system startup I open sumatrapdf from the taskbar to have it open all the remembered files from the previous session

**Reply by GitHubRulesOK** [reply]
exactly which version are you using but you should only be testing with SumatraPDF pre-release since only that version is normally "fix"able I am illustrating 3.6.15925 but should be same in 15927 (ok just reloaded latest and  it still works the same)

-reuse-instance : **tells an already open SumatraPDF to load the indicated files**. If there are several running instances, behavior is undefined. Only needed when communicating with SumatraPDF through DDE (use the ReuseInstance setting instead otherwise).

So if a tab is already open it should respect reuse and reopen the same tab. (DDE is active)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/3743695f-4bd4-4c7f-a89a-95956714461f)

 If a single window is open it usually is ignored as starting a fresh run via command line rather than DDE counts as a fresh instance.
Reuse instance = false
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/4a77a553-b066-46b9-9b01-083718098223)
Reuse instance = true
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/5a2a9f57-1e48-4ffe-80f4-4312a763e60b)

**Comment by HGSchuetz** [comment]
Hi,
thx for clarification. But how can I manage, to open each new PDF in the SAME Sumatra window? I only need ONE window, and each PDF should be displayed in the same Sumatra window (instance, program, task or how ever we want to call it). So, opening the nexte PDF should close the previous one.
If not possible now, a settings entry like  **SameWindow = true**  would be excellent! THX!
Best regards
Hans

**Comment by GitHubRulesOK** [comment]
Confused by request if you only want one window you would use the tabbed or non tabbed interface and close the unwanted one then open another file, this would be the same in most pdf readers such as Acrobat DC and Microsoft Edge etc when not using tabs

You are expecting command line to have a replacement function but command line open 2  files in Edge application/pdf or Acrobat will generally open 2 separate windows. (sorry Edge view is currently set to secured do not run pdf as a script. each of these was started as a command line without tabs active.

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f55eb0b9-a0df-4509-a780-63203b1bebc5)


Generically while a PDF is open in a window it is locked against outside changes thus in such cases you either close first file to free up the viewer OR open a second window and then background close the first for a better user experience without flicker.

SumatraPDF has a workaround that the others do not have and that is that if the files are under 32 MB they are left unlocked, which means you can use substitution thus copy file 1.pdf to %temp%\Under32MB.pdf and open %temp%\Under32Mb.pdf then copy file 2.pdf  to %temp%\Under32MB.pdf and it will auto refresh
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/77341ee3-f3e9-440d-8d68-04dffae5a792)

**Comment by HGSchuetz** [comment]
Thans a lot for your fast response!
Well, I have thousands of PDFs ... want to use Sumatra as a Viewer in my AHK app ... so that's where my request for only one window is coming from. I want to go fast thru tons of little PDFs.
But your hint with copying to temp sounds like a possible solution.
Will check this out!
BR
Hans
